torch/distributed/_sharded_tensor/__init__.py

```diff
@@ -453,31 +455,36 @@ def shard_parameter(
             placement=placement,
         )
         shards_metadata.append(shard_metadata)
+        # Reshape to get shard for each rank and we don't want autograd
+        # recording here for the narrow op and 'tensor_to_scatter' should be a
+        # leaf variable in the autograd graph.
+        tensor_to_scatter = tensor.narrow(
+            sharding_spec.dim,  # type: ignore[arg-type]
+            shard_metadata.shard_offsets[sharding_spec.dim],  # type: ignore[union-attr, arg-type, index]
+            shard_metadata.shard_sizes[sharding_spec.dim],  # type: ignore[union-attr, index]
+        ).clone().detach().contiguous()
+        tensors_to_scatter.append(
+            tensor_to_scatter
+        )
```

Comment:
Once your new sharding spec extension PR is ready, I guess the logic here can reuse what's been written in the shard api?

Reply:
yeah agreed, this might just be using `ChunkShardingSpec.shard`